### ISSUE TITLE
CLI: support passing a remote URL as file argument

### ISSUE THREAD

**Issue by shykes** [dagger/dagger#6957]
## Problem

When calling a Dagger Function from the CLI, I can pass the path of a local file as an argument of type `File`, but not the URL of a remote file. This would be very useful.

## Solution

Add support for remote URLs when passing arguments of type `File` from the CLI.

**Comment by helderco** [comment]
This has already been done in:
- https://github.com/dagger/dagger/pull/6769

**Comment by shykes** [comment]
No, I think this is different. #6769 allows passing a file from a git repo - which is indeed very useful. But I want to pass a regular http url to a regular file. And that unfortunately does not seem to work.

**Comment by helderco** [comment]
Oh, sorry! I actually read URL as Git! 😅 

That's a good idea, and easy to add 👍

**Comment by helderco** [comment]
Hmm... both `http://` and `https://` are valid git schemes. How would you know if you want `dag.Git` or `dag.Http`?

**Comment by helderco** [comment]
Some tools prefix the scheme with `git+`, like `git+https://github.com/...` ([example](https://hatch.pypa.io/latest/config/dependency/#supported-vcs)).

**Comment by jedevc** [comment]
FYI, this is related to https://github.com/dagger/dagger/pull/7609 as well, since it also has different schemes.

**Comment by helderco** [comment]
@guillaume, can you own this? It's easy to do but requires a decision on the scheme and consistency with the other changes you've been making.

**Comment by grouville** [comment]
> @grouville, can you own this? It's easy to do but requires a decision on the scheme and consistency with the other changes you've been making.

Yes sure 🙏 👌

**Comment by helderco** [comment]
Update: this is blocked on standardizing URI references (git, http, other vcs…) in:

* [https://github.com/dagger/dagger/pull/7708](https://github.com/dagger/dagger/pull/7708)
